packages/flutter/lib/src/cupertino/segmented_control.dart

```diff
@@ -0,0 +1,683 @@
+/// When one option in the segmented control is selected, the other options
+/// in the segmented control cease to be selected.
+///
+/// A segmented control can feature any [Widget] as one of its [children].
+/// All [children] must have a [Key] value that can be used to identify
+/// the widget. Typically, these keys will be instances of the [ValueKey]
+/// class.
+///
+/// When the state of the segmented control changes, the widget calls the
+/// [onValueChanged] callback. The [Key] of the currently selected widget
+/// is returned in the [onValueChanged] callback. Most widgets that use
+/// a segmented control will listen for the [onValueChanged] callback
+/// and rebuild the segmented control with a new [value] to update which
+/// option is currently selected.
+///
+/// The height of the segmented control is determined by the height of the
+/// tallest widget provided in the [children] list. The width of the segmented
+/// control is determined by the horizontal constraints on its parent. The
+/// available horizontal space is divided by the number of provided [children]
+/// to determine the width of each widget. The selection area for each of the
+/// [children] will then  be expanded to fill the calculated space, so each
+/// widget will appear to have the same dimensions.
+///
+/// See also:
+///
+/// * <https://developer.apple.com/design/human-interface-guidelines/ios/controls/segmented-controls/>
+class SegmentedControl extends StatefulWidget {
+  /// Creates an iOS-style segmented control bar.
+  ///
+  /// The [children] and [onValueChanged] arguments must not be null. Further,
+  /// the length of the [children] list must be greater than or equal to two.
+  ///
+  /// Each widget in the list of [children] must have a [Key] value that
+  /// uniquely identifies this widget. This [Key] is what will be returned
+  /// in the [onValueChanged] callback when a new child in the [children] list
+  /// is selected. Most widgets that use a segmented control will listen for
+  /// the [onValueChanged] callback and rebuild the segmented control with
+  /// the returned [Key] as the new [value] to update the currently selected
+  /// option.
+  ///
+  /// The [value] is the currently selected value for the segmented control.
+  /// If no [value] is provided, or the [value] is null, no widget will
+  /// appear as initially selected. This [value] must be the [Key] of one
+  /// of the widgets in the list of [children].
+  SegmentedControl({
+    Key key,
+    @required this.children,
+    @required this.onValueChanged,
+    this.value,
+  })  : assert(children != null),
+        assert(children.length >= 2),
+        assert(onValueChanged != null),
+        assert(() {
+          bool validKey = false;
+          for (Widget child in children) {
+            if (child.key == value) {
+              validKey = true;
+            }
+          }
+          return value == null || validKey;
```

Comment:
you can write this as `assert(value == null || children.any((Widget child) => child.key == value))` or some such